cli/command/image/build.go

```diff
@@ -250,16 +272,34 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		}
 	}
 
-	ctx := context.Background()
+	if options.stream && dockerfileCtx == nil {
```

Comment:
Is `dockerfileCtx == nil` the same as `!options.dockerfileFromStdin()` ?  Could we use the latter so it's more clear what this means?